### DISCUSSION TITLE
filer: skip uploading existing files

### DISCUSSION THREAD

**Discussion by zemul** [seaweedfs/seaweedfs#5626] [General]
If a file already exists in the filer and its metadata MD5 matches the MD5 of the newly uploaded file, then skipping the upload is feasible. If this approach is acceptable, I will submit a merge request.

**Comment by zemul** [comment]
@chrislusf

**Reply by chrislusf** [reply]
that means read the existing md5 first. The upload process would be a bit different. It's better to do it on the client side.
